viam-cartographer/src/carto_facade/map_builder.cc

```diff
@@ -102,15 +102,28 @@ bool MapBuilder::SaveMapToFile(bool include_unfinished_submaps,
 }
 
 void MapBuilder::AddSensorData(
+    const std::string &sensor_id,
     cartographer::sensor::TimedPointCloudData measurement) {
     trajectory_builder->AddSensorData(kRangeSensorId.id, measurement);
 }
 
-void MapBuilder::StartLidarTrajectoryBuilder() {
-    VLOG(1) << "MapBuilder::StartLidarTrajectoryBuilder";
-    trajectory_id = map_builder_->AddTrajectoryBuilder(
-        {kRangeSensorId}, trajectory_builder_options_,
-        GetLocalSlamResultCallback());
+void MapBuilder::AddSensorData(const std::string &sensor_id,
+                               cartographer::sensor::ImuData measurement) {
+    trajectory_builder->AddSensorData(kIMUSensorId.id, measurement);
+}
+
+void MapBuilder::StartTrajectoryBuilder(bool use_imu_data) {
+    VLOG(1) << "MapBuilder::StartTrajectoryBuilder";
+    if (use_imu_data) {
```

Comment:
[nit] make a sensor list in if statement then one line 
```
trajectory_id = map_builder_->AddTrajectoryBuilder(
            sensorList, trajectory_builder_options_,
            GetLocalSlamResultCallback());
    }
```
outside of if statement